core/src/error.rs

```diff
@@ -134,6 +136,31 @@ pub enum Error {
 	HttpNotImplemented,
 }
 
+impl<T> From<TokioSendError<T>> for Error {
+	fn from(_: TokioSendError<T>) -> Self {
+		// This error can originate during a `send` call on the client's
+		// `to_back: tokio::sync::mpsc::Sender<FrontToBack>` (channel to communicate between
+		// frontend -> backend).
+		//
+		// The tokio's `SendError<T>` is plainly a wrapper over T.
+		//
+		// The error wraps the value passed to `send`.
+		// Tokio's `send` can only fail if the receive half of the channel is closed
+		// (either `close` was called, or the receiver was dropped).
+		Error::Internal(InternalError::Disconnected)
+	}
+}
+
+impl From<futures_channel::mpsc::SendError> for Error {
+	fn from(err: futures_channel::mpsc::SendError) -> Self {
+		if err.is_full() {
```

Comment:
I think tokio hid the enum because they could add other states and require other checks in a non breaking change.

I wonder whether it's worth documenting/future-proofing this by being more explicit, like:

```rust
if err.is_full() { Full }
else if err.is_disconnected() { Disconnected }
else { Unknown }
```

what do you think?